packages/block-editor/src/components/block-popover/inbetween.js

```diff
@@ -91,66 +91,81 @@ function BlockPopoverInbetween( {
+					: null;
+				const nextRect = nextElement
+					? nextElement.getBoundingClientRect()
+					: null;
+
+				if ( isVertical ) {
+					if ( isRTL() ) {
+						return {
+							top: previousRect
+								? previousRect.bottom
+								: nextRect.top,
+							left: previousRect
+								? previousRect.right
+								: nextRect.right,
+							right: previousRect
+								? previousRect.left
+								: nextRect.left,
+							bottom: nextRect
+								? nextRect.top
+								: previousRect.bottom,
+							height: 0,
+							width: 0,
+						};
+					}
+
+					return {
+						top: previousRect ? previousRect.bottom : nextRect.top,
+						left: previousRect ? previousRect.left : nextRect.left,
+						right: previousRect
+							? previousRect.right
+							: nextRect.right,
+						bottom: nextRect ? nextRect.top : previousRect.bottom,
+						height: 0,
+						width: 0,
+					};
+				}
+
+				if ( isRTL() ) {
+					return {
+						top: previousRect ? previousRect.top : nextRect.top,
+						left: previousRect ? previousRect.left : nextRect.right,
+						right: nextRect ? nextRect.right : previousRect.left,
+						bottom: previousRect
+							? previousRect.bottom
+							: nextRect.bottom,
+						height: 0,
+						width: 0,
+					};
+				}
 
-		if ( isVertical ) {
-			if ( isRTL() ) {
 				return {
-					top: previousRect ? previousRect.bottom : nextRect.top,
-					left: previousRect ? previousRect.right : nextRect.right,
-					right: previousRect ? previousRect.left : nextRect.left,
-					bottom: nextRect ? nextRect.top : previousRect.bottom,
+					top: previousRect ? previousRect.top : nextRect.top,
+					left: previousRect ? previousRect.right : nextRect.left,
+					right: nextRect ? nextRect.left : previousRect.right,
+					bottom: previousRect
+						? previousRect.bottom
+						: nextRect.bottom,
 					height: 0,
 					width: 0,
-					ownerDocument,
 				};
-			}
-
-			return {
-				top: previousRect ? previousRect.bottom : nextRect.top,
-				left: previousRect ? previousRect.left : nextRect.left,
-				right: previousRect ? previousRect.right : nextRect.right,
-				bottom: nextRect ? nextRect.top : previousRect.bottom,
-				height: 0,
-				width: 0,
-				ownerDocument,
-			};
-		}
-
-		if ( isRTL() ) {
-			return {
-				top: previousRect ? previousRect.top : nextRect.top,
-				left: previousRect ? previousRect.left : nextRect.right,
-				right: nextRect ? nextRect.right : previousRect.left,
-				bottom: previousRect ? previousRect.bottom : nextRect.bottom,
-				height: 0,
-				width: 0,
-				ownerDocument,
-			};
-		}
-
-		return {
-			top: previousRect ? previousRect.top : nextRect.top,
-			left: previousRect ? previousRect.right : nextRect.left,
-			right: nextRect ? nextRect.left : previousRect.right,
-			bottom: previousRect ? previousRect.bottom : nextRect.bottom,
-			height: 0,
-			width: 0,
-			ownerDocument,
+			},
 		};
-	}, [ previousElement, nextElement ] );
+	}, [ previousElement, nextElement, isVisible ] );
```

Comment:
Added the `isVisible` dependency here, since the variable is used inside the hook